packages/qdrant-loader-core/src/qdrant_loader_core/llm/factory.py

```
@@ -95,6 +95,37 @@ def create_provider(settings: LLMSettings) -> LLMProvider:
+        except ImportError:
+            return _NoopProvider()
+
+    is_gemini = (
+        "gemini" in provider_name
+        or provider_name in ("google", "google_genai", "vertex", "vertexai")
+        or (
+            base_host is not None
+            and (
+                base_host == "generativelanguage.googleapis.com"
+                or base_host.endswith(".googleapis.com")
+                and "generativelanguage" in base_host
```

Ruff (0.15.14)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)